pkg/rke2/rke2_linux.go

```diff
@@ -199,3 +205,22 @@ func hostnameFQDN() (string, error) {
 
 	return strings.TrimSpace(b.String()), nil
 }
+
+func hostnameFromMetadataEndpoint() (string, error) {
+	resp, err := http.Get("http://169.254.169.254/latest/metadata/local-hostname")
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("unexpected status code: %d", resp.StatusCode)
```

Comment:
Is there anything more descriptive that we could put in here??

Reply:
I can put something more descriptive, but this error is never outputted to the user in the current implementation because if this function returns an error then we use `hostnameFQDN`.

Reply:
Let's remove this then if it's never going to happen.

Reply:
Updated.